packages/flutter_adaptive_scaffold/test/adaptive_scaffold_test.dart

```diff
@@ -239,6 +239,174 @@ void main() {
       });
     },
   );
+
+  testWidgets(
+    'when destinations passed with all data, it shall not be null',
+    (WidgetTester tester) async {
+      const List<NavigationDestination> destinations = <NavigationDestination>[
+        NavigationDestination(
+          icon: Icon(Icons.inbox_outlined),
+          selectedIcon: Icon(Icons.inbox),
+          label: 'Inbox',
+        ),
+        NavigationDestination(
+          icon: Icon(Icons.video_call_outlined),
+          selectedIcon: Icon(Icons.video_call),
+          label: 'Video',
+        ),
+      ];
+
+      await tester.pumpWidget(
+        const MaterialApp(
+          home: MediaQuery(
+            data: MediaQueryData(size: Size(700, 900)),
+            child: AdaptiveScaffold(
+              destinations: destinations,
+            ),
+          ),
+        ),
+      );
+
+      final Finder fNavigationRail = find.descendant(
+        of: find.byType(AdaptiveScaffold),
+        matching: find.byType(NavigationRail),
+      );
+      final NavigationRail navigationRail = tester.firstWidget(fNavigationRail);
+      expect(
+        navigationRail.destinations,
+        isA<List<NavigationRailDestination>>(),
+      );
+      expect(
+        navigationRail.destinations.length,
+        destinations.length,
+      );
+
+      for (final NavigationRailDestination element
```

Comment:
tiny nit: "destination" might be a better name, since element is an overloaded term here.

Reply:
I Will make change in some time and update the PR.